packages/button/src/Button.spec.tsx

```diff
@@ -78,6 +78,14 @@ describe('packages/button', () => {
       expect((button as HTMLButtonElement).type).toBe('submit');
     });
 
+    test(`renders a button as another component if the "as" prop is set`, () => {
+      const { container } = renderButton({
+        as: 'p',
+      });
+      expect(container.querySelector('button')).toBeNull();
+      expect(container.querySelector('p')).not.toBeNull();
+    });
+
```

Comment:
👍

Reply:
Shouldn't matter in this test, but you could check `element.tagname` in case there's a `p` element inside the button